src/evolutionary_keras/optimizers.py

```diff
@@ -353,10 +352,9 @@ def undo_flatten(self, flattened_weights):
         """
         new_weights = []
         for i, layer_shape in enumerate(self.shape):
-            flat_layer = flattened_weights[
-                self.length_flat_layer[i] : self.length_flat_layer[i]
-                + self.length_flat_layer[i + 1]
-            ]
+            start_index = sum(self.length_flat_layer[: i + 1])

```

Comment:
Should be `i` and not `i+1` right?

```suggestion
            start_index = sum(self.length_flat_layer[: i])
```

Reply:
No because enumarate starts at 0, and `length_flat_layer[: 0]=[]` while in that case we want the first element (which is a 0).

Reply:
Ah, and the first item is set to 0 above. Missed that, sorry. Then this is ready to go.